yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -573,10 +573,16 @@ object ApplicationMaster extends Logging {
   }
 
   private[spark] def sparkContextInitialized(sc: SparkContext): Unit = {
+    if (master == null){
```

Comment:
Yes, you could reach this, but the NPE at that point was just a side effect of a different bug. Fix that bug, NPE is gone. Remove the NPE, but leave the bug in, something else will blow up.